### PR TITLE
Added File Rename feature on Main Window (#5798) and Filename textbox on QuickTaskMenu

### PR DESCRIPTION
**Pull request by kadamiec**
Main features:
1. Allow to rename the file on disk from Main Window, available only when file exists. It also renames file in RecentTasks
2. Allow to rename(the default) filename in Quick Task Menu shown after taking screenshot

Other improvements:
- FileExistForm: Validations of filename (forbidden characters)
- FileExistForm: Default actions on keys: Enter -> UseNewFilename; Escape -> UseUniqueFilename
- Helpers: RenameFile - new params to allow overwrite file and show exception
- Helpers: IsValidFilePath - validate directory and allow to show exceptions

Screens:
ShareX-RenameFile
ShareX-QuickTaskMenu-FilenameEditable

### CODE REVIEW

**Comment by AngryCarrot789** [comment]
Not sure if you added this, but in the main form (with all the image previews), having the "Rename to..." button show in the context menu (when you right click an image) would be pretty cool too

**Comment by kadamiec** [comment]
Screenshot:
ShareX-QuickTaskMenu-FilenameEditable

In order to edit the filename immediately after capture (before saving), you need first to add "_Show Quick Task Menu_" in "_After Capture tasks_" so then you get this:
ShareX-QuickTaskMenu-FilenameEditable

**Comment by martinbra** [comment]
fixes #5798

**Comment by martinbra** [comment]
bump.

**Comment by martinbra** [comment]
Is there any timeline for this PR to be merged?

Em sáb., 4 de jun. de 2022 23:13, github-actions[bot] <
***@***.***> escreveu:

File: ShareX.HelpersLib/Helpers/Helpers.cs [hunk 1]
```diff
@@ -979,21 +979,68 @@ public static void CreateDirectoryFromFilePath(string filePath)
...
+                    || e is PathTooLongException
+                    || e is NotSupportedException)
+                {
+                    MessageBox.Show($"Provided path: {path} is invalid!", "ShareX - Path Validation Error " + Resources.Error, MessageBoxButtons.OK, MessageBoxIcon.Error);
+                }
+
             }
-            catch (ArgumentException) { }
-            catch (PathTooLongException) { }
-            catch (NotSupportedException) { }
```
Comment (botatooo) [inline ShareX.HelpersLib/Helpers/Helpers.cs:992]: Is there any reason for removing these error catchers?

Comment (martinbra) [reply ShareX.HelpersLib/Helpers/Helpers.cs:992]: @kadamiec ?

Comment (kadamiec) [reply ShareX.HelpersLib/Helpers/Helpers.cs:992]: They are not removed, just handled correctly together with a message few lines above, as there they would be silent.
Sorry for late response.

**Comment by martinbra** [comment]
bump.

**Pull request closed without merging**